### PR TITLE
feat: implement block device wipe

### LINKED ISSUE #9731: support wiping non-volume blockdevices
Required for Omni bare-metal infra provider. cc @utkuozdemir

### PR DESCRIPTION
**Pull request by smira**
Fixes #9731

The wipe doesn't require a reboot, but it requires the blockdevice not to be used as a volume.

### CODE REVIEW

**Review by frezbo — APPROVED** [review]

**Review by dsseng — APPROVED** [review]

File: cmd/talosctl/cmd/talos/wipe.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+		})
+	},
+}
+
+func wipeMethodValues() []string {
+	var method storage.BlockDeviceWipeDescriptor_Method
+
+	values := make([]string, method.Descriptor().Values().Len())
+
+	for idx := range method.Descriptor().Values().Len() {
```
Comment (DmitriyMV) [inline cmd/talosctl/cmd/talos/wipe.go:63]: Why not `for idx := range len(storage.BlockDeviceWipeDescriptor_Method_name)`?

File: internal/app/storaged/server.go [hunk 2]
```diff
@@ -7,23 +7,35 @@ package internal
...
 )
 
 // Server implements storage.StorageService.
 //
 // It is only kept here for compatibility purposes, proper API is to query `block.Disk` resources.
 type Server struct {
 	storage.UnimplementedStorageServiceServer
-	Controller runtime.Controller
+	Controller      runtime.Controller
+	MaintenanceMode bool
```
Comment (DmitriyMV) [inline internal/app/storaged/server.go:38]: It's always `true` any reason for this variable?

**Review by DmitriyMV — APPROVED** [review]

File: cmd/talosctl/cmd/talos/wipe.go (hunk 1 above)
Comment (smira) [reply cmd/talosctl/cmd/talos/wipe.go:63]: in theory it might have gaps (we don't have it, but that was my understanding)

**Pull request merged**